src/Microsoft.Graph.Core/Tasks/LargeFileUploadTask.cs

```diff
@@ -43,6 +43,10 @@ public LargeFileUploadTask(IParsable uploadSession, Stream uploadStream, int max
             {
                 throw new ArgumentException("Must provide stream that can read and seek");
             }
+            if (uploadStream.Length < 1)
```

Comment:
doesn't this put us at risk of getting a not supported exception if they are streaming from another http service for example without buffering in memory?
https://learn.microsoft.com/en-us/dotnet/api/system.io.stream.length?view=net-7.0#exceptions

Reply:
We do check before this that the stream is seekable on if block above. Which is the situation when the exception could be thrown. Is there another scenario where we expect the length to throw an error?

Reply:
The seekable constraint is also due to the need to split up the stream into smaller substreams. So that task can jump to indexes in the stream as it slices it up for upload.  
At the moment, if a stream is not seek able, the task wouldn't be able to slice it up.

Reply:
oh right, I forgot about the slicing. Is there a way people could tell us not to slice and just push the whole stream in a single request?

Reply:
That would be possible by passing a slice size that is the same size as the stream. 
The alternative would be to simply use the request builders that the task uses to make a put a request with the target stream as the payload.

Reply:
right, thanks for the additional context and disregard my comment then. I'm guessing it'll be an opportunity for improvement if we receive customer evidence for that. They'll have to buffer the stream in the meantime.